src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.Digest.cs

```diff
@@ -415,12 +415,28 @@ public static void AppendKeyValue(this StringBuilder sb, string key, string valu
             if (includeQuotes)
             {
                 sb.Append('"');
+                int lastSpecialIndex = 0, specialIndex;
+                while (true)
+                {
+                    specialIndex = value.IndexOfAny(new[] { '"', '\\' }, lastSpecialIndex);
```

Comment:
If we're concerned about avoiding allocations, we shouldn't be allocating a new char[] array for every iteration here.